Source/SwiftLintBuiltInRules/Rules/Style/VerticalWhitespaceOpeningBracesRule.swift

```diff
@@ -150,43 +180,371 @@ extension VerticalWhitespaceOpeningBracesRule: OptInRule {
+        }
+
+        private func isClosureSignatureIn(_ inToken: TokenSyntax) -> Bool {
+            // Check if the "in" token is part of a closure signature by looking at its parent
+            var currentNode = Syntax(inToken)
+            while let parent = currentNode.parent {
+                if parent.is(ClosureSignatureSyntax.self) {
+                    return true
+                }
+                // Stop traversing if we hit a different expression or declaration
+                if parent.is(ExprSyntax.self) || parent.is(DeclSyntax.self) {
+                    break
+                }
+                currentNode = parent
+            }
+            return false
+        }
+
+        private func checkForViolationAfterToken(_ token: TokenSyntax) {
+            // We analyze the trivia of the token immediately following the token
+            if let nextToken = token.nextToken(viewMode: .sourceAccurate) {
+                let triviaAnalysis = analyzeTriviaForViolations(
+                    trivia: nextToken.leadingTrivia,
+                    position: token.endPositionBeforeTrailingTrivia
+                )
+
+                if let violation = triviaAnalysis {
+                    violations.append(
+                        ReasonedRuleViolation(
+                            position: violation.position,
+                            correction: .init(
+                                start: violation.position,
+                                end: violation.endPosition,
+                                replacement: ""
+                            )
+                        )
+                    )
+                }
+            }
+        }
+
+        private func analyzeTriviaForViolations(
+            trivia: Trivia,
+            position: AbsolutePosition
+        ) -> (position: AbsolutePosition, endPosition: AbsolutePosition)? {
+            let analysis = analyzeTrivia(trivia: trivia, startPosition: position)
+
+            // Only flag violations if we found an empty line that wasn't allowed
+            guard let startPos = analysis.violationStartPosition,
+                  let endPos = analysis.violationEndPosition,
+                  analysis.hasViolation else {
+                return nil
+            }
+
+            return (position: startPos, endPosition: endPos)
+        }
+
+        private func analyzeTrivia(
+            trivia: Trivia,
+            startPosition: AbsolutePosition
+        ) -> TriviaAnalysis {
+            var result = TriviaAnalysis()
+            var currentPosition = startPosition
+
+            for piece in trivia {
+                let positionAdvance = result.processTriviaPiece(piece, currentPosition: currentPosition)
+                currentPosition = currentPosition.advanced(by: positionAdvance)
+            }
+
+            return result
+        }
+    }
+
+    final class Rewriter: ViolationsSyntaxRewriter<SeverityConfiguration<VerticalWhitespaceOpeningBracesRule>> {
```

Comment:
The rewriter is currently not used at all since the visitor does the replacement already and the rule doesn't specify `explicitRewriter: true`.